src/components/TeacherComponents/UnitsLearningThemeFilters/UnitsLearningThemeFilters.tsx

```diff
@@ -25,56 +26,77 @@ export type UnitsLearningThemeFiltersProps = {
   trackingProps?: LearningThemeSelectedTrackingProps;
+  idSuffix: string;
 };
+
 const UnitsLearningThemeFilters = ({
-  labelledBy,
   learningThemes = [],
   selectedThemeSlug,
   linkProps,
   trackingProps,
+  idSuffix,
 }: UnitsLearningThemeFiltersProps) => {
   const [skipFiltersButton, setSkipFiltersButton] = useState(false);
-  const listStateProps = useCategoryFilterList({
-    selectedKey: selectedThemeSlug,
-    getKey: (
-      linkProps: UnitListingLinkProps | SpecialistUnitListingLinkProps,
-    ) => {
-      if (linkProps.search?.["learning-theme"]) {
-        return linkProps.search?.["learning-theme"];
-      } else return "all";
-    },
-  });
-
-  const learningThemesMapped = learningThemes
+  const learningThemesMapped: Array<RadioTheme> = learningThemes
     ? learningThemes
         .map((learningTheme) => {
           return {
             label: learningTheme?.themeTitle,
             slug: learningTheme?.themeSlug,
           };
         })
-        .sort(
-          (
-            a: {
-              label: string | undefined | null;
-              slug: string | undefined | null;
-            },
-            b: {
-              label: string | undefined | null;
-              slug: string | undefined | null;
-            },
-          ) => {
-            if (a?.slug === "no-theme") {
-              return 0;
-            } else if (b?.slug === "no-theme") {
-              return -1;
-            } else {
-              return 0;
-            }
-          },
-        )
+        .filter((theme) => isRadioTheme(theme))
+        .map((theme) => theme as RadioTheme)
+        .sort((a: RadioTheme, b: RadioTheme) => {
+          if (a.slug === "no-theme") {
+            return 0;
+          } else if (b.slug === "no-theme") {
+            return -1;
+          } else {
+            return 0;
+          }
+        })
     : [];
 
+  const router = useRouter();
+  const { track } = useAnalytics();
+
+  // Manual setting the active theme slug on select to avoid a delay in visual state updating while the page reloads
+  const [activeThemeSlug, setActiveThemeSlug] = useState(selectedThemeSlug);
+
+  const [focussedThemeSlug, setFocussedThemeSlug] = useState<
+    string | undefined
+  >(undefined);
+
+  const onChange = (theme: { label: string; slug: string }) => {
+    setActiveThemeSlug(theme.slug);
+    if (trackingProps) {
+      const { keyStageSlug, subjectSlug } = trackingProps;
+
+      track.browseRefined({
+        platform: "owa",
+        product: "teacher lesson resources",
+        engagementIntent: "refine",
+        componentType: "filter_link",
+        eventVersion: "2.0.0",
+        analyticsUseCase: "Teacher",
+        filterType: "Learning theme filter",
+        filterValue: theme.label,
+        activeFilters: { keyStage: [keyStageSlug], subject: [subjectSlug] },
+      });
+    }
+
+    const query = theme.slug === "all" ? {} : { "learning-theme": theme.slug };
+    router.push(
```

Comment:
🏄🏻 I think this should be a `replace` and not a `push`. I wouldn't expect to be taken back to the last filter results if I hit the back button, instead i'd expect to go to the previous page.

Reply:
agreed!